lms/static/scripts/frontend_apps/components/FilePickerApp.js

```diff
@@ -34,6 +34,21 @@ import GroupConfigSelector from './GroupConfigSelector';
  * @prop {Error} error
  */
 
+/**
+ * Blackboard files are of a "url" content type, but the URL is opaque and
+ * not helpful to the user. In those cases, use a canned message, otherwise
+ * truncate the content URL for display.
+ *
+ * @param {string} url
+ * @returns {string}
+ */
+function formatContentURL(url) {
+  // All Blackboard file URLs start with the literal string `blackboard:`
+  if (url.includes('blackboard:')) {
```

Comment:
Maybe check if it starts with that? It should theoretically faster, the best type of fast!